examples/repo_example/succinct_hash_bin_delegations.py

```diff
@@ -0,0 +1,226 @@
+# based on the byte digest of the target file name. More precisely, only a
+# prefix of the target file name byte digest is needed to assign it to the
+# correct hash bin.
+#
+# In succinct hash bin delegation all delegated roles (bins) are represented by
+# one single "succinct_roles" dictionary or when a targets file is loaded with a
+# SuccinctRoles instance from tuf/api/metadata.py
+#
+# The number of bins, name prefix for all bins and key threshold are all
+# attributes that need to be configured.
+
+# Number of bins, bit length and bin number computation
+# -----------------
+# The right number of bins depends on the expected number of target files in a
+# repository. The maximum number of bins is 2 of the power of 32.
+# For the purpose of this example we choose:
+NUMBER_OF_BINS = 32
+#
+# The bit length is the number of bits that will be used to calculate the
+# corresponding bin for each target path and it's directly related to the
+# a number of bins:
+BIT_LENGTH = int(math.log2(NUMBER_OF_BINS))
+#
+# To calculate the appropriate bin number for a particular target the path to
+# that target is signed using a cryptographic algorithm (like "sha256") which
+# guarantees uniform distribution. After that, the first BIT_LENGTH of bits from
+# the byte digest are used to calculate the bin number in hexadecimal format.
+# For more information read SuccinctRoles implementation in tuf/api/metadata.py
+
+# Delegated role (bin) name format
+# -----------------
+# Each bin has a name in the format of f"{NAME_PREFIX}-{bin_number}".
+#
+# Name prefix is the common prefix of all delegated target roles (bins) in a
+# particular SuccinctRoles object. For our example it will be:
+NAME_PREFIX = "delegated_bin"
+#
+# The suffix "bin_number" is a zero-padded hexidentical number of that
+# particular bin.
+
+# Keys and threshold
+# -----------------
+# Given that the primary concern of succinct hash bin delegation is to reduce
+# network overhead it was decided that all bins will be signed by the same
+# set of one or more signing keys.
+#
+# Before generating the keys a decision has to be made about the number of keys
+# required for signing the bins or in other words the value of the threshold.
+# Threshold is a number with the same meaning as defined in the spec:
+# https://theupdateframework.github.io/specification/latest/#threshold
+#
+# For the purpose of this example we choose the threshold to be:
+THRESHOLD = 1
+
+
+def create_key() -> Tuple[Key, SSlibSigner]:
+    """Generates a new Key and Signer."""
+    sslib_key = generate_ed25519_key()
+    return Key.from_securesystemslib_key(sslib_key), SSlibSigner(sslib_key)
+
+
+# Now, we can generate a THRESHOLD number of keys:
+succinct_roles_keys: List[Key] = []
+succinct_roles_signers: List[SSlibSigner] = []
+for i in range(0, THRESHOLD):
+    key, signer = create_key()
+    succinct_roles_keys.append(key)
+    succinct_roles_signers.append(signer)
```

Comment:
The loop will only have one iteration. You can reduce complexity by removing the loop. And, for the sake of this example, use only one key.

Reply:
I wanted to give an option to set a different threshold.
If you set a different threshold than 1 it will no longer work.